src/main/java/TaskTracker.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class TaskTracker {
+    private TreeMap<Integer, Task> tasks;
```

Comment:
Great use of appropriate data structures that helps with scalibility. Keep it up!